ansible/roles/test/tasks/fdb.yml

```diff
@@ -73,3 +73,85 @@
   always:
     - name: clear FDB table
       command: sonic-clear fdb all
+
+- block:
+  - name: clear FDB table
+    command: sonic-clear fdb all
+
+  - name: Get ports in portchannels
+    set_fact:
+      portchannel_members_all: []
+  - set_fact:
+      portchannel_members_all: "{{ portchannel_members_all + item['members'] }}"
+    with_items: "{{ minigraph_portchannels.values() }}"
+
+  - name: Get DUT ports info
+    set_fact:
+      vlan_ports: "{%for port in minigraph_vlans['Vlan1000']['members'] %}{{minigraph_port_indices[port]}}{%if not loop.last%} {%endif%}{%endfor%}"
+      lag_ports: "{%for port in portchannel_members_all %}{{minigraph_port_indices[port]}}{%if not loop.last%} {%endif%}{%endfor%}"
+
+  - name: Initialize variables
+    set_fact:
+      dummy_mac_prefix: "02:11:22:33"
+
+  - name: "Set ptf facts"
+    set_fact:
+      ptf_test_name: FDB test
+      ptf_test_dir: ptftests
+      ptf_test_path: fdb_test.FdbConfigReloadTest
+      ptf_platform: remote
+      ptf_platform_dir: ptftests
+      ptf_test_params:
+        - testbed_type='{{ testbed_type }}'
+        - vlan_ports='{{ vlan_ports }}'
+        - lag_ports='{{ lag_ports }}'
+        - router_mac='{{ ansible_interface_facts['Ethernet0']['macaddress'] }}'
+        - dummy_mac_prefix='{{ dummy_mac_prefix }}'
+      ptf_extra_options: "--relax --debug info --log-file /tmp/fdb_test.FdbTest.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}.log "
+
+  - debug: msg="ptf --test-dir {{ ptf_test_dir }} {{ ptf_test_path }} {% if ptf_qlen is defined %} --qlen={{ ptf_qlen }} {% endif %} {% if ptf_platform_dir is defined %} --platform-dir {{ ptf_platform_dir }} {% endif %} --platform {{ ptf_platform }} {% if ptf_test_params is defined %} -t \"{{ ptf_test_params | default([]) | join(';') }}\" {% endif %} {{ ptf_extra_options | default(\"\")}} 2>&1"
+
+  - name: Run PTF script to inject packets with different src MAC
+    shell: ptf --test-dir {{ ptf_test_dir }} {{ ptf_test_path }} {% if ptf_qlen is defined %} --qlen={{ ptf_qlen }} {% endif %} {% if ptf_platform_dir is defined %} --platform-dir {{ ptf_platform_dir }} {% endif %} --platform {{ ptf_platform }} {% if ptf_test_params is defined %} -t "{{ ptf_test_params | default([]) | join(';') }}" {% endif %} {{ ptf_extra_options | default("")}} 2>&1
+    args:
+      chdir: /root
+    delegate_to: "{{ ptf_host }}"
+    async: 300
+    poll: 0
+
+  - name: Reload config
+    command: config reload -y
+    become: yes
+
+  - name: Wait until the PTF script completed execution
+    shell: "ps -ef | grep /usr/bin/ptf | grep -v grep"
+    register: ptf_script_state
+    retries: 60
+    until: ptf_script_state.rc == 1
+    delay: 5
+    delegate_to: "{{ ptf_host }}"
+    ignore_errors: yes
+
+  - name: Wait some extra time to ensure that all FDB entries are in DB
+    pause: seconds=10
+
+  - name: Count FDB entries from SDK
+    shell: "docker exec syncd sx_api_fdb_dump.py | grep {{ dummy_mac_prefix }} | wc -l"
```

Comment:
>grep {{ dummy_mac_prefix }} [](start = 51, length = 27)

In stead of testing only dummy mac, how about testing all AGEABLE mac?

Reply:
In this new test case, PTF script doesn't inject packets with src MAC set to MAC of PTF ports. So, all AGEABLE MAC are the dummy MAC addresses.

Reply:
I understand that all AGEABLE MAC are the dummy MAC addresses. So the two option are the effectively the same.

I am wondering the test purpose. Should we test all AGEABLE(dynamic) mac, no matter whether the MAC is generated with dummy mac or not?

---
In reply to: [278421108](https://github.com/Azure/sonic-mgmt/pull/887#discussion_r278421108) [](ancestors = 278421108)

Reply:
Got your point. Updated the script to count all AGEABLE MAC.